File: RepoTransBench/target_projects/Python/C++/GoogleCloudPlatform_localllm/public_tests/test_public_modelfiles_full.cpp
#include <gtest/gtest.h>
#include <string>
#include <vector>
#include <algorithm>

namespace {

std::vector<std::string> public_find_llmfiles(const std::string&) {
    // Simulate test directory contents
    return {"lion.Q4_0.gguf", "giraffe.Q8_0.gguf"};
}

bool is_llmfile(const std::string& fname) {
    return fname.size() > 5 && fname.substr(fname.size()-5) == ".gguf";
}

TEST(TestPublicModelFilesFull, PublicFindLlmFiles) {
    std::string test_dir = "/tmp/some-llm-model-dir";
    auto output = public_find_llmfiles(test_dir);
    ASSERT_TRUE(std::any_of(output.begin(), output.end(), [](const std::string& f){return f.find(".gguf") != std::string::npos;}));
    std::string joined;
    for (const auto& x : output) joined += x;
    ASSERT_NE(joined.find("lion.Q4_0.gguf"), std::string::npos);
}

TEST(TestPublicModelFilesFull, PublicIsLlmfile) {
    ASSERT_TRUE(is_llmfile("rhino.Q7_0.gguf"));
    ASSERT_FALSE(is_llmfile("zebra.txt"));
}

} // namespace

File: RepoTransBench/target_projects/Python/Rust/GoogleCloudPlatform_localllm/public_tests/test_public_modelfiles_full.rs
#[test]
fn test_public_find_llmfiles() {
    let test_dir = "/tmp/some-llm-model-dir";
    let test_files = vec!["lion.Q4_0.gguf", "giraffe.Q8_0.gguf"];
    let mut found_gguf = false;
    for f in &test_files {
        if f.ends_with(".gguf") {
            found_gguf = true;
        }
    }
    assert!(found_gguf);
    let content = test_files.join("");
    assert!(content.contains("lion.Q4_0.gguf"));
}

#[test]
fn test_public_is_llmfile() {
    let filename1 = "rhino.Q7_0.gguf";
    let filename2 = "zebra.txt";
    assert!(filename1.ends_with(".gguf"));
    assert!(!filename2.ends_with(".gguf"));
}